lib/datasets/zip_writer.rb

```diff
@@ -1,12 +1,42 @@
+require 'zip'
+require 'digest'
 
 # Writes a zip file from a volume source path
 # to a destination path.
 class ZipWriter
-
   # Write the dataset from src_path to
-  # the directory at dest_path.  
+  # the directory at dest_path.
   # @param [Pathname] src_path
   # @param [Pathname] dest_path
-  def write(src_path, dest_path)
+  # @yield [input_zip, output_zip] optional block for implementation of copying
+  # from input zip to output zip. By default gets all txt files in input zip
+  # and copies them to output zip.
+  def write(src_path, dest_path, &block)
+    tmp_path = dest_path.dirname.join(Dir::Tmpname.make_tmpname('dataset', '.zip'))
```

Comment:
I think this is a good approach to avoid calling `#open` twice on the same file as we were when using Tempfile.